src/SFML/Graphics/Shader.cpp

```diff
@@ -37,25 +37,141 @@
+            GLEXT_glActiveTexture = (void (GL_FUNCPTR *)(GLenum))sf::Context::getFunction("glActiveTextureARB");
+            GLEXT_glClientActiveTexture = (void (GL_FUNCPTR *)(GLenum))sf::Context::getFunction("glClientActiveTextureARB");
+
+            if (!GLEXT_glActiveTexture ||
+                !GLEXT_glClientActiveTexture)
+                GLEXT_multitexture = false;
+        }
+
+        GLEXT_shading_language_100 = sf::Context::isExtensionAvailable("GL_ARB_shading_language_100");
+
+        GLEXT_shader_objects = sf::Context::isExtensionAvailable("GL_ARB_shader_objects");
+
+        if (GLEXT_shader_objects)
+        {
+            GLEXT_glAttachObject = (void (GL_FUNCPTR *)(GLhandleARB, GLhandleARB))sf::Context::getFunction("glAttachObjectARB");
+            GLEXT_glCompileShader = (void (GL_FUNCPTR *)(GLhandleARB))sf::Context::getFunction("glCompileShaderARB");
+            GLEXT_glCreateProgramObject = (GLhandleARB (GL_FUNCPTR *)())sf::Context::getFunction("glCreateProgramObjectARB");
+            GLEXT_glCreateShaderObject = (GLhandleARB (GL_FUNCPTR *)(GLenum))sf::Context::getFunction("glCreateShaderObjectARB");
+            GLEXT_glDeleteObject = (void (GL_FUNCPTR *)(GLhandleARB))sf::Context::getFunction("glDeleteObjectARB");
+            GLEXT_glGetHandle = (GLhandleARB (GL_FUNCPTR *)(GLenum))sf::Context::getFunction("glGetHandleARB");
+            GLEXT_glGetInfoLog = (void (GL_FUNCPTR *)(GLhandleARB, GLsizei, GLsizei *, GLcharARB *))sf::Context::getFunction("glGetInfoLogARB");
+            GLEXT_glGetObjectParameteriv = (void (GL_FUNCPTR *)(GLhandleARB, GLenum, GLint *))sf::Context::getFunction("glGetObjectParameterivARB");
+            GLEXT_glGetUniformLocation = (GLint (GL_FUNCPTR *)(GLhandleARB, const GLcharARB *))sf::Context::getFunction("glGetUniformLocationARB");
+            GLEXT_glLinkProgram = (void (GL_FUNCPTR *)(GLhandleARB))sf::Context::getFunction("glLinkProgramARB");
+            GLEXT_glShaderSource = (void (GL_FUNCPTR *)(GLhandleARB, GLsizei, const GLcharARB **, const GLint *))sf::Context::getFunction("glShaderSourceARB");
+            GLEXT_glUniform1f = (void (GL_FUNCPTR *)(GLint, GLfloat))sf::Context::getFunction("glUniform1fARB");
+            GLEXT_glUniform1i = (void (GL_FUNCPTR *)(GLint, GLint))sf::Context::getFunction("glUniform1iARB");
+            GLEXT_glUniform2f = (void (GL_FUNCPTR *)(GLint, GLfloat, GLfloat))sf::Context::getFunction("glUniform2fARB");
+            GLEXT_glUniform3f = (void (GL_FUNCPTR *)(GLint, GLfloat, GLfloat, GLfloat))sf::Context::getFunction("glUniform3fARB");
+            GLEXT_glUniform4f = (void (GL_FUNCPTR *)(GLint, GLfloat, GLfloat, GLfloat, GLfloat))sf::Context::getFunction("glUniform4fARB");
+            GLEXT_glUniformMatrix4fv = (void (GL_FUNCPTR *)(GLint, GLsizei, GLboolean, const GLfloat *))sf::Context::getFunction("glUniformMatrix4fvARB");
+            GLEXT_glUseProgramObject = (void (GL_FUNCPTR *)(GLhandleARB))sf::Context::getFunction("glUseProgramObjectARB");
```

Comment:
> Also why the C casts?

Couldn't be bothered with all the reinterpret_casts since I knew this was a draft.

> Also is it really necessary to check all these extensions/functions? 

Yes.

> Isn't there any "if we've got this, we've got all others"?

No.

> Also wouldn't it be possible (or even better) to use some macro magic to merge these lines together with the definitions above, while putting everything in some easy to read function style line?

A working macro would not reduce the amount of code that has to be typed anyway and probably make it look even uglier.